hack/ensure-deps/ensure-docker.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+# Copyright 2022 VMware Tanzu Community Edition contributors. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+# This script installs dependencies needed for running build-tce.sh and deploy-tce.sh
+
+BUILD_OS=$(uname -s)
+export BUILD_OS
+# Make sure docker is installed
+echo "Checking for Docker..."
+if [[ -z "$(command -v docker)" ]]; then
+    echo "Installing Docker..."
+    if [[ "$BUILD_OS" == "Linux" ]]; then
+        curl -fsSL https://get.docker.com -o get-docker.sh
+        DRY_RUN=1 sh ./get-docker.sh
```

Comment:
Was `DRY_RUN=1` added for testing? That should be removed now, right?

Reply:
Yeah for getting logs as it will be used hardly we can remove this.